### PR TITLE
Update godep.md

### PR DESCRIPTION
**Pull request by cblecker**
Update with proper hack commands and procedures to update godep.md

### CODE REVIEW

**Comment by cblecker** [comment]
I've tested this personally, as well as had another user test them [here](https://github.com/kubernetes/kubernetes/pull/44082#issuecomment-292698640).

**Comment by cblecker** [comment]
cc @sttts who has helped previously with the godep scripts

File: contributors/devel/godep.md [hunk 1]
```diff
@@ -69,24 +69,19 @@ godep get $DEP/...
...
 # state.
-rm -rf $KPATH/src/$DEP # repo root
-godep get $DEP/...
-# Change code in Kubernetes, if necessary.
-rm -rf Godeps
-rm -rf vendor
+cd $KPATH/src/$DEP # repo root
+git checkout <$REVISION/$TAG/$BRANCH>
+cd $KPATH/src/k8s.io/kubernetes
+godep update $DEP/...
```
Comment (sttts) [inline contributors/devel/godep.md]: Will this work if `$DEP` depends on one of our staging repos? Like k8s.io/metrics.

Comment (sttts) [inline contributors/devel/godep.md]: Why do we need this update call after all? Does it speed up the following godep-save? Can we leave vendor/ and Godeps/Godeps.json when running godep-save.sh?

Comment (cblecker) [inline contributors/devel/godep.md]: Godeps/Godeps.json doesn't pick up the intention to upgrade without it. And yes, with this procedure, I'm not blowing away vendor/ or Godeps/

Comment (sttts) [inline contributors/devel/godep.md]: I am still not convinced that we have to bring godep-update into play here if we call `./hack/godep-save.sh` afterwards. Therefore my question again: does it speed up `./hack/godep-save.sh` considerably?

Also note that we pin the godep version in godep-save/restore.sh. If we call godep-update directly, it's not necessarily the pinned version.

Comment (cblecker) [inline contributors/devel/godep.md]: The second point you make is excellent -- you're absolutely right that this doesn't work if we're pinning the version.

Let me test this out some more, and I'll report back with findings. Appreciate the review, @sttts!

**Comment by cblecker** [comment]
I'm going to close these as these scripts really just need to be improved rather than worked around.

**Pull request closed without merging**